CMakeLists.txt

```diff
@@ -10,14 +10,14 @@ endif (POLICY CMP0048)
 project(googletest-distribution)
 set(GOOGLETEST_VERSION 1.10.0)
 
+
 if (CMAKE_VERSION VERSION_LESS "3.1")
-  add_definitions(-std=c++11)
+  add_definitions(-std=c++11 -D_POSIX_C_SOURCE=200809L)
 else()
   set(CMAKE_CXX_STANDARD 11)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
-  if(NOT CYGWIN AND NOT MSYS)
-    set(CMAKE_CXX_EXTENSIONS OFF)
-  endif()
+  set(CMAKE_CXX_EXTENSIONS OFF)
```

Comment:
Is this a behavior change for CYGWIN and MSYS?

Reply:
Yes, I am actually unsure how this would compile outside MSYS and CYGWIN as well (or are posix extensions not used on other systems?). Compiling with extensions disabled also disables posix functionality. This commit enables the extensions that you need (posix) and not more. This makes it all as compiler agnostic as possible. Pure C++ is of course even nicer but not very practical.

Even nicer would be to put a define in the cpp files where you use them, but this is not really feasible without a biggish refactor (and provides limited advantage).

Reply:
Thanks for this. I think this has follow-on effects for a large enough collection of our user base that we're unable to accept this change.

Reply:
I understand a bit. I don't understand why gtest compiles without extensions at all under Linux. For example fork() and isatty() are not part of the standard but are POSIX extensions. Ref http://man7.org/linux/man-pages/man3/isatty.3.html and http://man7.org/linux/man-pages/man2/fork.2.html . As these and other calls are used throughout the project you should define somewhere which POSIX standard you are using. 

I do understand as I did not test it on every system, but still this change should have no impact as it only enables some features. It does not disable them such as the change adding ```set(CMAKE_CXX_EXTENSIONS OFF)``` which broke the build on at least CYGWIN, MINGW and QNX.

I will make a PR to fix QNX.